### PR TITLE
Feature/rul 44

### PR DESCRIPTION
**Pull request by milkovsky**

### CODE REVIEW

File: src/Form/ReactionRuleAddForm.php [hunk 1]
```diff
@@ -81,10 +91,36 @@ public function form(array $form, FormStateInterface $form_state) {
...
+   *
+   * @param $event_name
+   *   The event base name.
+   * @param array $configuration
+   *   The event configuration.
+   *
+   * @return \Drupal\rules\Core\RulesConfigurableEventHandlerInterface|null
+   *   The event handler, null if there is no proper handler.
+   */
+  protected function getEventHandler($event_name, $configuration = []) {
```
Comment (milkovsky) [inline src/Form/ReactionRuleAddForm.php:129]: not sure if it is a good place for this method

File: src/Core/RulesEventManager.php [hunk 2]
```diff
@@ -64,7 +64,7 @@ public function processDefinition(&$definition, $plugin_id) {
    * Gets the base name of a configured event name.
    *
    * For a configured event name like {EVENT_NAME}--{SUFFIX}, the base event
-   * name {EVENT_NAME} is returned.
+   * name {EVENT_NAME} is returned.s
```
Comment (fago) [inline src/Core/RulesEventManager.php]: ?

File: src/EventHandler/ConfigurableEventHandlerEntityBundle.php [hunk 3]
```diff
@@ -51,7 +105,8 @@ public function validate() {
    * {@inheritdoc}
    */
   public function getEventNameSuffix() {
-    // Nothing to do by default.
+    $parts = explode(':', $this->pluginId);
```
Comment (fago) [inline src/EventHandler/ConfigurableEventHandlerEntityBundle.php]: Would be good to comment the non-obvious, That's the bundle. Why not just return $this->configuration['bundle']; instead?

File: src/EventHandler/ConfigurableEventHandlerEntityBundle.php [hunk 4]
```diff
@@ -16,6 +16,40 @@
...
+   * @var string
+   */
+  protected $entityType;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityType = $this->getEventNameSuffix();
```
Comment (fago) [inline src/EventHandler/ConfigurableEventHandlerEntityBundle.php]: that's weird, the suffix should be the bundle not?

File: src/Form/ReactionRuleAddForm.php (hunk 1 above)
Comment (fago) [inline src/Form/ReactionRuleAddForm.php:129]: seems fine. but it should explain that there is only a handler when the event is configurable

File: src/Form/ReactionRuleAddForm.php [hunk 5]
```diff
@@ -81,10 +91,36 @@ public function form(array $form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function save(array $form, FormStateInterface $form_state) {
+    $event_name = $form_state->getValue('events')[0]['event_name'];
```
Comment (fago) [inline src/Form/ReactionRuleAddForm.php]: This changes should be in buildEntity()

File: src/Form/ReactionRuleAddForm.php [hunk 6]
```diff
@@ -71,9 +72,18 @@ public function form(array $form, FormStateInterface $form_state) {
       '#title' => $this->t('React on event'),
       '#options' => $options,
       '#required' => TRUE,
+      '#ajax' => $this->getDefaultAjax(),
       '#description' => $this->t('Whenever the event occurs, rule evaluation is triggered.'),
     ];
 
+    $form['event_configuration'] = array();
+    if ($values = $form_state->getValue('events')) {
```
Comment (fago) [inline src/Form/ReactionRuleAddForm.php:81]: This should be read from the entity. You might need to make sure the entity in form state gets updated on ajax submission.

File: src/Form/ReactionRuleAddForm.php [hunk 7]
```diff
@@ -71,9 +72,18 @@ public function form(array $form, FormStateInterface $form_state) {
       '#title' => $this->t('React on event'),
       '#options' => $options,
       '#required' => TRUE,
+      '#ajax' => $this->getDefaultAjax(),
       '#description' => $this->t('Whenever the event occurs, rule evaluation is triggered.'),
```
Comment (fago) [inline src/Form/ReactionRuleAddForm.php:76]: Let's add executes submit callback here as in \Drupal\Core\Render\Element\Submit and add       '#submit' => array('::submitForm'), as \Drupal\Core\Entity\EntityForm::actions , this should update the entity in form state correctly.

File: src/EventHandler/ConfigurableEventHandlerEntityBundle.php (hunk 4 above)
Comment (milkovsky) [inline src/EventHandler/ConfigurableEventHandlerEntityBundle.php]: damn... it should be base name. so we do not need getEventNameSuffix, but getEventBaseName instead, right?

we get bundle from form state.

I do not like that getEventBaseName is in the rules event manager. I can not call it easily from here... What would you suggest?

File: src/Form/ReactionRuleAddForm.php (hunk 7 above)
Comment (milkovsky) [inline src/Form/ReactionRuleAddForm.php:76]: Drupal\Core\Render\Element\Submit does not have any #submit

File: src/EventHandler/ConfigurableEventHandlerEntityBundle.php (hunk 4 above)
Comment (fago) [inline src/EventHandler/ConfigurableEventHandlerEntityBundle.php]: hm, not easy. Else, I could see it being a ConfigurableEventTrait which has the helper. The classes that need it, can use the helper then. We should mention the trait from the configurable event interface then though, e.g. via @see

File: src/Form/ReactionRuleAddForm.php (hunk 7 above)
Comment (fago) [inline src/Form/ReactionRuleAddForm.php:76]: #executes_submit_callback or so I meant, sry

File: src/EventHandler/ConfigurableEventHandlerEntityBundle.php [hunk 8]
```diff
@@ -16,6 +16,40 @@
...
+   * @var string
+   */
+  protected $entityType;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityType = $this->getEventNamePrefix();
```
Comment (milkovsky) [inline src/EventHandler/ConfigurableEventHandlerEntityBundle.php]: Not sure about this. I need entity type here.

File: src/Core/RulesConfigurableEventHandlerInterface.php [hunk 9]
```diff
@@ -90,6 +90,7 @@ public function extractConfigurationFormValues(array &$form, FormStateInterface
   public function validate();
 
   /**
+   * todo
```
Comment (milkovsky) [inline src/Core/RulesConfigurableEventHandlerInterface.php]: I replaced getEventNameSuffix with getEventNamePrefix. Is it ok?

If yes, I will change the comment

**Comment by milkovsky** [comment]
outdated

**Pull request closed without merging**